src/createStore.js

```diff
@@ -163,38 +172,31 @@ export default function createStore(reducer, preloadedState, enhancer) {
    * return something else (for example, a Promise you can await).
    */
   function dispatch(action) {
-    if (!isPlainObject(action)) {
-      throw new Error(
-        'Actions must be plain objects. ' +
-        'Use custom middleware for async actions.'
-      )
-    }
-
-    if (typeof action.type === 'undefined') {
-      throw new Error(
-        'Actions may not have an undefined "type" property. ' +
-        'Have you misspelled a constant?'
-      )
-    }
-
-    if (isDispatching) {
-      throw new Error('Reducers may not dispatch actions.')
-    }
+    validateAction(action)
+    currentState = store.final.reducer(currentState, action)
+    store.final.onChange()
```

Comment:
I still agree with #580, this behaviour is actually what makes a library like redux-batched-subscribe (RBS) possible, if the action was passed to subscription listeners this would make it difficult to perform any debounce or batching logic.

Keeping consistent with this API was my main reason not implementing the feature in RBS yet. But based on the number of requests to support conditional batching logic based on the action + the majority of the forks existing just to add this feature, I am re-evaluating things.

Does passing the action "internally" within redux to onChange run into the same/similar issues #580 is trying to avoid by passing the action to external subscription listeners?